source/material_model/rheology/diffusion_dislocation.cc

```diff
@@ -88,99 +123,40 @@ namespace aspect
-
-                        const double dislocation_prefactor = 0.5 * std::pow(dislocation_creep_parameters.prefactor,-1.0/dislocation_creep_parameters.stress_exponent);
-                        const double dislocation_strain_rate_dependence = std::pow(dislocation_strain_rate, (1.-dislocation_creep_parameters.stress_exponent)/dislocation_creep_parameters.stress_exponent);
-                        const double dislocation_T_and_P_dependence = std::exp(std::max(dislocation_creep_parameters.activation_energy + pressure*dislocation_creep_parameters.activation_volume,0.0)/
-                                                                               (dislocation_creep_parameters.stress_exponent*constants::gas_constant*temperature));
-
-                        const double dislocation_viscosity = std::min(std::max(dislocation_prefactor * dislocation_strain_rate_dependence
-                                                                               * dislocation_T_and_P_dependence,
-                                                                               minimum_viscosity), maximum_viscosity);
-
-                        diffusion_strain_rate = dislocation_viscosity / (diffusion_viscosity + dislocation_viscosity) * edot_ii;
-                        dislocation_strain_rate = diffusion_viscosity / (diffusion_viscosity + dislocation_viscosity) * edot_ii;
-
-                        ++stress_iteration;
-                        AssertThrow(stress_iteration < stress_max_iteration_number,
-                                    ExcMessage("No convergence has been reached in the loop that determines "
-                                               "the ratio of diffusion/dislocation viscosity. Aborting! "
-                                               "Residual is " + Utilities::to_string(log_strain_rate_residual) +
-                                               " after " + Utilities::to_string(stress_iteration) + " iterations. "
-                                               "You can increase the number of iterations by adapting the "
-                                               "parameter 'Maximum strain rate ratio iterations'."));
-
-                        log_strain_rate_residual = std::abs(std::log(diffusion_strain_rate/old_diffusion_strain_rate));
-                        stress_ii = 2.0 * edot_ii * 1./(1./diffusion_viscosity + 1./dislocation_viscosity);
-                      }
-                    while (log_strain_rate_residual > log_strain_rate_residual_threshold);
-
-                    break;
-                  }
-              }
-
-            // The effective viscosity, with minimum and maximum bounds
+
+            // KINSOL works on vectors and so the scalar (log) stress is inserted into
+            // a vector of length 1
+            Vector<double> log_stress_ii(1);
+            log_stress_ii[0] = std::log(stress_ii);
```

Comment:
This is your starting guess. How is it chosen? Based on the current log stress? It might be worth trying the output of the previous loop iteration, under the assumption that what you found in the previous iteration is a good starting guess for the current iteration. Not sure whether that's true, since you're not looping over quadrature points, for example, but a thought.